File: validation/duplication_d0_20260818/corpus/structural/structural.go
package structural

func baseline(left int, right int) int {
    first := transform(
        left, right, left + right, 1,
    )
    second := combine(
        first, left * 2, right * 3, 2,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

func exactCopy(left int, right int) int {
    first := transform(
        left, right, left + right, 1,
    )
    second := combine(
        first, left * 2, right * 3, 2,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

func whitespaceCommentCopy(left int, right int) int {
    first := transform(
        left, right, left + right, 1,
    )
    // formatting/comment only
    second := combine(
        first, left * 2, right * 3, 2,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

func renamed(x int, y int) int {
    alpha := transform(
        x, y, x + y, 1,
    )
    beta := combine(
        alpha, x * 2, y * 3, 2,
    )
    gamma := finalize(
        alpha, beta, x - y, 3,
    )
    return publish(
        alpha, beta, gamma, 4,
    )
}

func literalChanged(left int, right int) int {
    first := transform(
        left, right, left + right, 11,
    )
    second := combine(
        first, left * 2, right * 3, 22,
    )
    third := finalize(
        first, second, left - right, 33,
    )
    return publish(
        first, second, third, 44,
    )
}

func operatorChanged(left int, right int) int {
    first := transform(
        left, right, left - right, 1,
    )
    second := combine(
        first, left * 2, right * 3, 2,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

func statementAdded(left int, right int) int {
    first := transform(
        left, right, left + right, 1,
    )
    extra := transform(right, left, right + left, 0)
    second := combine(
        first, left * 2, right * 3, 2,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

func statementRemoved(left int, right int) int {
    first := transform(
        left, right, left + right, 1,
    )
    second := combine(
        first, left * 2, right * 3, 2,
    )
    return publish(
        first, second, third, 4,
    )
}

func statementReordered(left int, right int) int {
    second := combine(
        first, left * 2, right * 3, 2,
    )
    first := transform(
        left, right, left + right, 1,
    )
    third := finalize(
        first, second, left - right, 3,
    )
    return publish(
        first, second, third, 4,
    )
}

File: validation/duplication_d0_20260818/corpus/structural/structural.ts
function baseline(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 1,
    );
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}

function exactCopy(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 1,
    );
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}

function whitespaceCommentCopy(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 1,
    );
    // formatting/comment only
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}

function renamed(x: number, y: number): number {
    const alpha = transform(
        x, y, x + y, 1,
    );
    const beta = combine(
        alpha, x * 2, y * 3, 2,
    );
    const gamma = finalize(
        alpha, beta, x - y, 3,
    );
    return publish(
        alpha, beta, gamma, 4,
    );
}

function literalChanged(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 11,
    );
    const second = combine(
        first, left * 2, right * 3, 22,
    );
    const third = finalize(
        first, second, left - right, 33,
    );
    return publish(
        first, second, third, 44,
    );
}

function operatorChanged(left: number, right: number): number {
    const first = transform(
        left, right, left - right, 1,
    );
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}

function statementAdded(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 1,
    );
    const extra = transform(right, left, right + left, 0);
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}

function statementRemoved(left: number, right: number): number {
    const first = transform(
        left, right, left + right, 1,
    );
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    return publish(
        first, second, third, 4,
    );
}

function statementReordered(left: number, right: number): number {
    const second = combine(
        first, left * 2, right * 3, 2,
    );
    const first = transform(
        left, right, left + right, 1,
    );
    const third = finalize(
        first, second, left - right, 3,
    );
    return publish(
        first, second, third, 4,
    );
}
